include/boost/http_proto/serializer.hpp

```diff
@@ -293,48 +338,147 @@ class BOOST_SYMBOL_VISIBLE
+     */
     using buffers_type =
         buffers::mutable_buffer_pair;
 
+    /**
+        Returns the remaining available capacity.
+
+        The returned value represents the available free
+        space in the backing fixed-sized buffers used by the
+        serializer associated with this stream.
+
+        The capacity is absolute and does not do any
+        accounting for any octets required by a chunked
+        transfer encoding.
+    */
     BOOST_HTTP_PROTO_DECL
     std::size_t
     capacity() const noexcept;
 
+    /**
+        Returns the number of octets serialized by this
+        stream.
+
+        The associated serializer stores stream output in its
+        internal buffers. The stream returns the size of this
+        output.
+    */
     BOOST_HTTP_PROTO_DECL
     std::size_t
     size() const noexcept;
 
+    /**
+        Returns a boolean indicating if the stream can
+        receive more input.
+
+        The fixed-sized buffers maintained by the associated
+        serializer can be sufficiently full from previous
+        calls to \ref stream::commit.
+
+        This function can be called to determine if the user
+        should drain the serializer via \ref serializer::consume calls
+        before attempting to fill the buffer sequence
+        returned from \ref stream::prepare.
+    */
     BOOST_HTTP_PROTO_DECL
     bool
     is_full() const noexcept;
 
+    /**
+        Returns a MutableBufferSequence capable of storing
+        input from the user
+
+        The returned buffer sequence is as wide as is
+        possible. If a non-chunked transfer encoding is
+        being used than the returned sequence encompasses
+        the unused area of the serializer's fixed-sized
+        buffers.
+
+        If a chunked transer encoding is used then space is
+        reserved for the chunk header in addition to the
+        closing CRLF required for the chunk data. In
+        addition to this, space is also reserved for the
+        last-chunk.
+
+        This is done so that users can chain calls to
+        \ref stream::commit and \ref stream::close
+        without having to drain the serializer via
+        \ref serializer::consume calls.
+
+        \exception std::length_error Thrown if the stream
+        has insufficient capacity and a chunked transfer
+        encoding is being used
+    */
     BOOST_HTTP_PROTO_DECL
     buffers_type
     prepare() const;
 
+    /**
+        Serialize and commit `n` bytes.
+
+        Once the sequence returned from \ref prepare has been
+        filled, the input can be serialized and committed to the
+        associated serializer's output area via a call to `commit(n)`.
+
+        If a chunked transfer encoding is being used then commit
+        is responsible for writing the chunk-header and also the
+        closing CRLF for the chunk-data. `n` denotes the size
+        of the chunk.
+
+        \exception std::logic_error Thrown if commit is
```

Comment:
I do prefer the at sign `@` instead of `\` but we should be consistent. Now the project is using both, which complicates full-text searches.